input/input.go

```diff
@@ -274,9 +294,27 @@ func (h *HtlcSucceedInput) CraftInputScript(signer Signer, txn *wire.MsgTx,
 	desc.InputIndex = txinIdx
 	desc.PrevOutputFetcher = prevOutputFetcher
 
-	witness, err := SenderHtlcSpendRedeem(
-		signer, &desc, txn, h.preimage,
+	isTaproot := txscript.IsPayToTaproot(desc.Output.PkScript)
+
+	var (
+		witness wire.TxWitness
+		err     error
 	)
+	if isTaproot {
+		if desc.ControlBlock == nil {
```

Comment:
Good q, needs to revisit this, but with the current code path, this is always called from the state read on disk which always includes the control block. Only in tests is it not set. 

I think I added this earlier on to find a bug: was able to make it fail early, rather than having a control block verification error down the line which is larger to debug.